src/Activation/NotificationPermissions.spec.tsx

```diff
@@ -0,0 +1,134 @@
+import React from "react"
+import { render, fireEvent, waitFor } from "@testing-library/react-native"
+import NotificationPermissions from "./NotificationPermissions"
+
+import {
+  PermissionsContext,
+  PermissionsContextState,
+} from "../PermissionsContext"
+import { OnboardingContext, OnboardingContextState } from "../OnboardingContext"
+import { PermissionStatus } from "../permissionStatus"
+
+describe("NotificationPermissions", () => {
+  describe("when a user enables notifications,", () => {
+    it("requests permissions", () => {
+      const onboardingProviderValue = createOnboardingProviderValue(jest.fn())
+
+      const notificationRequestSpy = jest.fn()
+      const permissionsProviderValue = createPermissionProviderValue(
+        notificationRequestSpy,
+      )
+
+      const { getByLabelText } = render(
+        <OnboardingContext.Provider value={onboardingProviderValue}>
+          <PermissionsContext.Provider value={permissionsProviderValue}>
+            <NotificationPermissions />
+          </PermissionsContext.Provider>
+        </OnboardingContext.Provider>,
+      )
+
+      fireEvent.press(getByLabelText("Enable Notifications"))
+
+      expect(notificationRequestSpy).toHaveBeenCalled()
+    })
+    it("marks onboarding complete", async () => {
```

Comment:
Are all of these actions asynchronous? I believe that if we are testing an action from the users we could potentially leave all this tests without the `async`?(probably wrong, but maybe worth the shot)

Reply:
Ah, so we can remove the `async` from the last test, but it needs to stay here because when a user enables notifications that request is async, so we have to wait for the `completeOnboardingSpy` to have been called _after_ that.